http/client.go

```diff
@@ -14,46 +14,63 @@ import (
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
-// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+type ClientOption func(*http.Client) error
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
-}
+// New creates a new http.Client with the Passed Transport Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
+// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
+func New(opts ...ClientOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
+	client := &http.Client{
+		Transport: t,
+	}
 
-	if CABundleFile != "" || CABundleDir != "" {
-		certs, err := getCertPool(CABundleFile, CABundleDir)
-		if err != nil {
+	for _, opt := range opts {
+		if err := opt(client); err != nil {
 			return nil, err
 		}
+	}
+
+	return client, nil
+}
 
-		t.TLSClientConfig = &tls.Config{
-			RootCAs: certs,
+func WithCABundle(CABundleFile, CABundleDir string) ClientOption {
+	return func(c *http.Client) error {
+		if CABundleFile != "" || CABundleDir != "" {
+			certs, err := getCertPool(CABundleFile, CABundleDir)
+			if err != nil {
+				return err
+			}
+			c.Transport.(*http.Transport).TLSClientConfig.RootCAs = certs
+		}
+		return nil
+	}
+}
+
+func WithAcceptInvalidHostname(CABundleFile, CABundleDir, acceptInvalidHostname string) ClientOption {
+	return func(c *http.Client) error {
+		err := WithCABundle(CABundleFile, CABundleDir)(c)
```

Comment:
We can probably "alias" the transport to avoid the verbosity

```suggestion
		transport := c.Transport.(*http.Transport)
		err := WithCABundle(CABundleFile, CABundleDir)(c)
```